diff --git a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
index 59f3d9c..701fad1 100644
--- a/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
+++ b/kirigami-plasmadesktop-integration/plasmadesktoptheme.cpp
@@ -51,15 +51,7 @@ public:
         connect(qGuiApp, &QGuiApplication::fontDatabaseChanged, this, &StyleSingleton::notifyWatchersConfigurationChange);
         qGuiApp->installEventFilter(this);
 
-        // NativeTextRendering is still distorted sometimes with fractional scale factors
-        // Given Qt disables all hinting with native rendering when any scaling is used anyway
-        // we can use Qt's rendering throughout
-        // QTBUG-126577
-        if (qApp->devicePixelRatio() == 1.0) {
-            QQuickWindow::setTextRenderType(QQuickWindow::NativeTextRendering);
-        } else {
-            QQuickWindow::setTextRenderType(QQuickWindow::QtTextRendering);
-        }
+        QQuickWindow::setTextRenderType(QQuickWindow::NativeTextRendering);
     }
 
     void refresh()
